src/main/java/bio/terra/service/snapshot/flight/delete/PerformAzureDatasetDependencyStep.java

```diff
@@ -0,0 +1,22 @@
+package bio.terra.service.snapshot.flight.delete;
+
+import bio.terra.service.job.OptionalStep;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+
+public class PerformAzureDatasetDependencyStep extends OptionalStep {
+  public PerformAzureDatasetDependencyStep(Step step) {
+    super(step);
```

Comment:
It occurs to me that we could get the same effect by wrapping one optional step in another, something like:
```
    super(new PerformDatasetStep(new PerformAzureStep(step)));
```